accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,233 @@
+      var classList = node.className;
+      classList = classList.split(' ');
+      var canAdd = true;
+      for (var i = 0; i < classList.length; i++) {
+        if (classList[i] == classText) {
+          canAdd = false;
+        }
+      }
+      //add class if it doesn't
+      if (canAdd) {
+        if (classList.length) {
+          node.className += ' ' + classText;
+        } else {
+          node.className += classText;
+        }
+      }
+    },
+    setId: function(block) {
+      if (this.isTopBlock) {
+        return this.parentId + '-node0';
+      }
+      return this.treeService.createId(block);
+    },
+    getInfo: function(block) {
+      // List all inputs
+      if (this.infoBlocks[block.id]) {
+        this.infoBlocks[block.id].length = 0;
+      } else {
+        this.infoBlocks[block.id] = [];
+      }
+
+      var blockInfoList = this.infoBlocks[block.id];
+
+      for (var j = 0, field; field = block.fieldRow[j]; j++) {
+        blockInfoList.push(field);
+      }
+
+      return this.infoBlocks[block.id];
+    },
+    getInputTypeLabel: function(connection) {
+      if (connection.check_) {
+        return connection.check_.join(', ').toUpperCase();
+      } else {
+        return 'any';
+      }
+    },
+    sendToSelected: function(block) {
+      if (this.sharedClipboardService) {
+        this.sharedClipboardService.pasteToMarkedConnection(block);
+        block.dispose(true);
+        alert('block sent to marked spot');
+      }
+    },
+    inputMenuSelected: function(connection, event) {
+      switch (event.target.value) {
+        case 'MARK_SPOT':
+          this.sharedClipboardService.markConnection(connection);
+          blocklyApp.debug && console.log('marked spot');
+          break;
+        case 'PASTE':
+          this.sharedClipboardService.paste(connection);
+          break;
+      }
+      event.target.selectedIndex = 0;
+    },
+    clipboardCompatibilityHTMLText: function(connection) {
+      if (this.sharedClipboardService.isConnectionCompatibleWithClipboard(
+              connection)) {
+        // undefined will result in the 'paste' option being ENABLED
+        return undefined;
+      } else {
+        // any value returned will result in the 'paste' option being DISABLED
+        return 'blockly-disabled';
+      }
+    },
+    valueOrStatement: function(inputBlock) {
+      if (inputBlock.type == Blockly.NEXT_STATEMENT) {
+        return 'statement';
+      } else {
+        return 'value';
+      }
+    },
+    markedBlockCompatibilityHTMLText: function(block) {
+      if (this.sharedClipboardService
+          .isBlockCompatibleWithMarkedConnection(block)) {
+        // undefined will result in the 'copy to marked block'
+        // option being ENABLED
+        return undefined;
+      } else {
+        // any value returned will result in the 'copy to marked block'
+        // option being DISABLED
+        return 'blockly-disabled';
+      }
+    }
```

Comment:
Multiple services is fine, as long as each is clearly named and has clearly-defined responsibilities. Definitely better than duplication.